packages/cloud_firestore_odm/lib/src/firestore_reference.dart

```diff
@@ -81,39 +81,6 @@ abstract class FirestoreDocumentReference<Model,
     batch.delete(reference);
   }
 
-  /// Sets data on the document, overwriting any existing data. If the document
-  /// does not yet exist, it will be created.
-  ///
-  /// If [SetOptions] are provided, the data can be merged into an existing
-  /// document instead of overwriting.
-  Future<void> set(Model model, [SetOptions? setOptions]) {
```

Comment:
Could we not use `update` for this instead? As it supports field values.

Reply:
I need to set the field values when creating the document. If I set the document then updated it that would count as 2 writes.

Reply:
You should be able to directly call `update` without calling `set` first. `update` is an upsert

Reply:
Based on the documentation in the admin SDK I'm not sure that's true

```dart
  /// Updates fields in the document referred to by this `DocumentReference`.
  /// The update will fail if applied to a document that does not exist.
  ///
  /// Nested fields can be updated by providing dot-separated field path
  /// strings.
  ///
  /// @param data An object containing the fields and values with which to
  /// update the document.
  /// @param precondition A Precondition to enforce on this update.
  /// @throws Error If the provided input is not valid Firestore data.
  /// @return A Promise resolved with the write time of this update.
  external JSPromise<WriteResult> update(
    JSObject data, [
    Precondition precondition,
  ]);
```

Also you would lose the type safety of creating the document with the model object since you could accidentally forget to set some non-nullable fields.